pymc_experimental/printing.py

```diff
@@ -0,0 +1,164 @@
+import itertools
+
+import numpy as np
+
+from pymc import Model
+from pymc.printing import str_for_dist, str_for_potential_or_deterministic
+from pytensor.compile.sharedvalue import SharedVariable
+from pytensor.graph.type import Constant
+from rich.box import SIMPLE_HEAD
+from rich.table import Table
+
+
+def _extract_value(var: SharedVariable | Constant) -> np.ndarray:
+    if isinstance(var, SharedVariable):
+        return var.get_value(borrow=True)
+    else:
+        return var.data
+
+
+def model_table(
```

Comment:
I would suggest splitting out the functionality for actually creating the string representation from this function. That intermediate representation could return a list of (name, expression, dims) tuples, which could then be fed into different types of string representation (of which `model_table` is one)

Reply:
Yeah I was thinking the same

Reply:
I refactored the utilities that convert model info (variable expressions, dims) so they can be used elsewhere